docs: Updates for Meltano Cloud GitHub App permissions and remove Alpha references (#7585)
meltano/meltano: docs/src/_cloud/security.md

1. Meltano will never store your secrets in clear text.
1. Meltano engineers do not have access to directly decrypt your secrets.
1. The decryption key for project secrets will never leave AWS servers.
1. Our IAM policies only allow the `decrypt` action within containers that are running project workloads or for services that require specific secrets to perform tasks such as sending notifications to your webhook urls.
1. Meltano uses envelope encryption strategy with AWS KMS keys to encrypt your secrets.

##### What encryption algorithms are used?

> The algorithm for encrypting secrets is an RSA symmetric encryption, which is a 256-bit AES-GCM encryption key.

Reviewer: ```suggestion
The algorithm for encrypting secrets is an AES symmetric encryption, using 256-bit AES-GCM encryption keys.
```
